### DISCUSSION TITLE
Blade Components - Better Scoped Slots

### DISCUSSION THREAD

**Discussion by Pennywise96** [laravel/framework#51757] [Ideas]
I wanted to ask what you all would think about this idea:

Currently you can access public methods or properties via the $component variable:

```
<x-blog.posts>
  {{ $component->category->name  }}
  @foreach($component->posts as $post)
      {{ $post->title }}
  @endforeach
</x-blog.posts>
```

But I would find it better/cleaner if the public methods or properties were immediately available:

```
<x-blog.posts as="posts">
  {{ $category->name  }}
  @foreach($posts as $post)
      {{ $post->title }}
  @endforeach
</x-blog.posts>
```

What do you think of this idea?

**Comment by ghost** [comment]
Think of this:
I send to view in blade 

[
'a' => (object)['aa' => 1],
'b' => (object)['aa' => 1],
]

$aa will not be unique anymore.
